libopenage/pathfinding/pathfinder.h

```diff
@@ -226,6 +234,13 @@ class PortalNode : public std::enable_shared_from_this<PortalNode> {
 	 * Priority queue node that contains this path node.
 	 */
 	heap_t::element_t heap_node;
+
+
+	using neighbours_t = std::unordered_map<portal_id_t, int>; //I wanted to use weak_ptr as key here but couldnt get it to compile
+
+	std::unordered_map<sector_id_t, neighbours_t> exits;
```

Comment:
It might be better/faster to not nest the unordered maps, i.e. to make two separate unordered maps.

- One mapping to get the exits (sector ID -> portal ID) :  `std::unordered_map<sector_id_t, portal_id_t>`
- Another mapping to get the distance cost to each exit (portal ID -> cost): `std::unordered_map<portal_id_t, int>`

This will result in another lookup in the second map, but it will probably still be faster since the maps will be closer to each other in memory.

Reply:
I think there is an issue with the second map.

say we have portals 1 and 2 that both connect sectors A and B, the distance between portals 1 and 2 may vary depending on which sector we enter from. 

That was part of the thinking behind this change.

Reply:
How would this handle portals, which connect the same two sectors? Maybe I'm missing something.

Reply:
I thought about this again and I think this is something we can just improve on later.